src/fetch.ts

```diff
@@ -2,17 +2,28 @@ import type { RpcGraphQL } from '@solana/rpc-graphql';
 
 export async function gql(...params: Parameters<RpcGraphQL['query']>) {
     const [source, variableValues] = params;
-    const res = await fetch('/api/graphql', {
+    return await postRequest('/api/graphql', { source, variableValues });
+}
+
+export async function _fetch(url: string, body: object) {
+    return await postRequest(url, body);
+}
+
+async function postRequest(url: string, body: object) {
+    const res = await fetch(url, {
         body: JSON.stringify({
-            source,
-            variableValues,
+            ...body,
```

Comment:
I think you can just do `JSON.stringify(body)` here, can't you?

Reply:
By doing that we'll get the following payload as body:

```json
"body": {
  "body": {"source":"abc", "variableValues": "xyz"}
}
```

Reply:
Oh my bad. I didn't remove the `{}` in JSON.stringify.

Reply:
Fixed it.